src/python/WMCore/MicroService/MSOutput/RelValPolicy.py

```diff
@@ -23,44 +24,60 @@ class RelValPolicyException(WMException):
 class RelValPolicy():
     """
     This module will contain the RelVal output data placement policy, where
-    destinations will be decided according to the dataset datatier.
+    destinations will be decided according to the dataset datatier and the
+    container lifetime will be decided based on the sample type (pre-release
+    or not).
 
     It's supposed to hold a policy driven by dataset datatier, and it's
     data structure looks like:
     [{"datatier": "tier_1", "destinations": ["rse_name_1", "rse_name_2"]},
      {"datatier": "tier_2", "destinations": ["rse_name_2"]},
      {"datatier": "default", "destinations": ["rse_name_3"]}]
 
+    The lifetime policy data structure is something like:
+    [{"releaseType": "pre", "lifetimeSecs": 120},
+     {"releaseType": "default", "lifetimeSecs": 360}]
+
     the 'default' key matches the case where a datatier is not specified
     in the policy.
     """
 
-    def __init__(self, policyDesc, listDatatiers, listRSEs, logger=None):
+    def __init__(self, tierPolicy, lifetimeDesc, listDatatiers, listRSEs, logger=None):
         """
         Given a policy data structure - as a list of dictionaries - it
         will validate the policy, the datatiers and RSEs defined in it,
         and it will convert the policy into a flat dictionary for easier
         data lookup.
-        :param policyDesc: list of dictionary items with the output rules
+        :param tierPolicy: list of dictionary items with the output rules
+        :param lifetimeDesc: list of dictionary items with the output
+            lifetime rules
         :param listDatatiers: flat list of existent datatiers in DBS
         :param listRSEs: flat list of existent Disk RSEs in Rucio
         :param logger: logger object, if any
         """
-        self.origPolicy = deepcopy(policyDesc)
+        self.origTierPolicy = deepcopy(tierPolicy)
+        self.origLifetimePolicy = deepcopy(lifetimeDesc)
 
         self.logger = getMSLogger(verbose=False, logger=logger)
 
-        self._validatePolicy(policyDesc, listDatatiers, listRSEs)
-        self.dictPolicy = self._convertPolicy(policyDesc)
+        self._validateTierPolicy(tierPolicy, listDatatiers, listRSEs)
+        self.tierPolicy = self._convertTierPolicy(tierPolicy)
+
+        self._validateLifetimePolicy(lifetimeDesc)
+        self.lifeTPolicy = self._convertLifePolicy(lifetimeDesc)
+
+        # regex to match against CMSSW pre-releases only, e.g.: CMSSW_1_2_3_pre12
+        self.preRegex = re.compile(r'CMSSW(_\d+){3}_pre(\d+)$')
 
     def __str__(self):
         """
         Stringify this object, printing the original policy
         """
-        objectOut = dict(originalPolicy=self.origPolicy, mappedPolicy=self.dictPolicy)
+        objectOut = dict(originalTierPolicy=self.origTierPolicy, mappedTierPolicy=self.tierPolicy,
```

Comment:
here again I see inconsistency with naming conventions used above, i.e. in MSOutput. Over there you had lowercase `lifetime`, while in workfkow it is CamelCase with capital letter, and here the dict will be created with camelCase with lowercase first letter. I don't know if it bring any problems, but honestly I totally lost which convention is used in python objects, in data exposed by services and APIs.